### PR TITLE
feat(vue): support for controlled props added to vue-searchbox

### PR DESCRIPTION
**Pull request by anik-ghosh-au7**
Description - 

 - Support for controlled behaviour added to vue-searchbox
 - The following props and event emitters were modified 
   - `value`
   - `onChange`
   - `onBlur`
   - `onKeyPress`
   - `onKeyDown`
   - `onkeyUp`
   - `onFocus`
 - Notion Card link - https://www.notion.so/appbase/SearchBox-Controlled-Behavior-90bb72a9296348b694921e8eb00504b2
 - Docs PR link - https://github.com/appbaseio/Docs/pull/190

demo - 
https://www.loom.com/share/6fc1c1a583af492282ec1559fcbac109

### CODE REVIEW

File: packages/vue-searchbox/src/components/SearchBox.jsx [hunk 1]
```diff
@@ -53,7 +54,10 @@ const SearchBox = {
 		URLParams: VueTypes.bool,
 		// RS API properties
 		id: VueTypes.string.isRequired,
-		value: VueTypes.any,
+		value: {
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchBox.jsx]: @anik-ghosh-au7 It should be `VueTypes.string.def('undefined')`

File: packages/vue-searchbox/src/components/SearchBox.jsx [hunk 2]
```diff
@@ -526,6 +541,13 @@ const SearchBox = {
 			event.stopPropagation();
 			event.preventDefault();
 		},
+		withTriggerQuery(eventName, event) {
+			const eventNameInKebabCase = convertToKebabCase(eventName);
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchBox.jsx]: @anik-ghosh-au7 You don't need this utility. We can just emit the events name in kebab case only.

File: packages/vue-searchbox/src/utils/helper.js [hunk 3]
```diff
@@ -206,6 +206,16 @@ export function parseFocusShortcuts(focusShortcutsArray) {
 	return parsedFocusShortcutsArray;
 }
 
+// used to convert camelCase event names to kebab-case
+export function convertToKebabCase(eventName) {
```
Comment (bietkul) [inline packages/vue-searchbox/src/utils/helper.js]: @anik-ghosh-au7 Remove it.

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 Docs PR is missing

File: packages/vue-searchbox/src/components/SearchBox.jsx [hunk 4]
```diff
@@ -53,7 +53,7 @@ const SearchBox = {
 		URLParams: VueTypes.bool,
 		// RS API properties
 		id: VueTypes.string.isRequired,
-		value: VueTypes.any,
+		value: VueTypes.string.def('undefined'),
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchBox.jsx]:
```suggestion
		value: VueTypes.string.def(undefined),
```

File: packages/vue-searchbox/src/components/SearchBox.jsx [hunk 5]
```diff
@@ -779,16 +793,16 @@ const SearchBox = {
 														on: getInputEvents({
 															onInput: this.onInputChange,
 															onBlur: e => {
-																this.$emit('blur', e);
+																this.withTriggerQuery('blur', e);
 															},
 															onFocus: this.handleFocus,
 															onKeyPress: e => {
-																this.$emit('keyPress', e);
+																this.withTriggerQuery('keyPress', e);
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchBox.jsx]:
```suggestion
																this.withTriggerQuery('key-press', e);
```

File: packages/vue-searchbox/src/components/SearchBox.jsx [hunk 6]
```diff
@@ -779,16 +793,16 @@ const SearchBox = {
...
 															},
 															onKeyDown: e =>
 																this.handleKeyDown(e, highlightedIndex),
 															onKeyUp: e => {
-																this.$emit('keyUp', e);
+																this.withTriggerQuery('keyUp', e);
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchBox.jsx]:
```suggestion
																this.withTriggerQuery('key-up', e);
```

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 It seems you didn't test the changes. Check the suggestions.
How can you set the default value as `undefined` string?
Share a demo

**Comment by anik-ghosh-au7** [comment]
demo - https://loom.com/share/f8fc3c17952647d3859f3f08c1cb025d

**Pull request merged**